addons/sale/models/sale_order_line.py

```diff
@@ -289,12 +289,19 @@ def _compute_no_variant_attribute_values(self):
                 if ptav._origin not in valid_values:
                     line.product_no_variant_attribute_value_ids -= ptav
 
-    @api.depends('product_id')
+    @api.depends('product_id', 'invoice_lines.move_id.state', 'invoice_lines.move_id.payment_state')
```

Comment:
No no no no no.

We do not want to reset the description of all order lines anytime something happens on the invoice...
The line must be created with the correct name and only be modified when needed, in the trigger business flow (invoice cancellation, ...)